src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,274 @@
+     * @param s String separated by white spaces
+     * @return List of String
+     */
+    private static List<String> convertToList(String s) {
+        String[] queries = s.trim().split("\\s+");
+        List<String> queryList = Arrays.asList(queries);
+        return queryList;
+    }
+
+    /**
+     * Parses each line of String in data file into Task.
+     * @param s Line of String in data file
+     * @return Task
+     * @throws DukeException if String cannot be parsed due to invalid
+     *     format or invalid date
+     */
+    public static Task parseFile(String s) throws DukeException {
+        String[] q = s.trim().split(",>");
+        try {
+            switch (q[0]) {
+            case "deadline":
+                return new Deadline(q[1], q[2].equals("X"), LocalDate.parse(q[3]));
+            case "event":
+                return new Event(q[1], q[2].equals("X"), LocalDate.parse(q[3]), LocalDate.parse(q[4]));
+            case "todo":
+                return new ToDo(q[1], q[2].equals("X"));
+            default:
+                throw new DukeException("Error parsing file data");
+            }
+        } catch (DateTimeException e) {
+            throw new DukeException("Error parsing date in file");
+        }
+    }
+
+    /**
+     * Parses user query of deadline into a Deadline instance.
+     * @param queryList User query list
+     * @return Deadline instance
+     * @throws DukeException if no deadline description, invalid format, or invalid date
+     */
+    private static Deadline parseUserDeadline(List<String> queryList) throws DukeException {
+        if (queryList.size() < 2) {
+            throw new DukeException("The description of a deadline cannot be empty.");
+        }
+        String name = "";
+        String deadlineString = "";
+        int byIndex = -1;
+        for (int i = 1; i < queryList.size(); i++) {
+            if (queryList.get(i).equals("/by")) {
+                byIndex = i;
+                continue;
+            }
+            if (byIndex == -1) {
+                name += i > 1 ? " " : "";
+                name += queryList.get(i);
+            } else {
+                deadlineString += i > byIndex + 1 ? " " : "";
+                deadlineString += queryList.get(i);
+            }
+        }
+        if (deadlineString.equals("")) {
+            throw new DukeException(emptyDeadlineDate);
+        }
+        try {
+            LocalDate deadline = LocalDate.parse(deadlineString);
+            return new Deadline(name, deadline);
+        } catch (DateTimeException e) {
+            throw new DukeException(invalidDate);
+        }
+    }
+
+    /**
+     * Parses user query of event into an Event instance.
+     * @param queryList User query list
+     * @return Event instance
+     * @throws DukeException if no event description, invalid format, or invalid date
+     */
+    private static Event parseUserEvent(List<String> queryList) throws DukeException {
```

Comment:
This method is slightly long and unclear. Perhaps you could add a comment or two to explain what you are doing here. 😃